wp-includes/service-workers.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Service Worker functions.
+ *
+ * @since ?
+ *
+ * @package PWA
+ */
+
+/**
+ * Initialize $wp_service_workers if it has not been set.
+ *
+ * @global WP_Service_Workers $wp_service_workers
+ * @return WP_Service_Workers WP_Service_Workers instance.
+ */
+function wp_service_workers() {
+	global $wp_service_workers;
+	if ( ! ( $wp_service_workers instanceof WP_Service_Workers ) ) {
+		$wp_service_workers = new WP_Service_Workers();
+	}
+	return $wp_service_workers;
+}
+
+/**
+ * Register a new service worker.
+ *
+ * @since ?
+ *
+ * @param string $handle   Name of the service worker. Should be unique.
+ * @param string $callback Callback method or relative path of the service worker.
```

Comment:
The type here can be `string|callable` with the param then being `$src`, where if it `is_callable()` then the return value is then used as the _source_ when printing.